src/build-data/botan-config.cmake.in

```diff
@@ -0,0 +1,71 @@
+set(_Botan_supported_components
+%{for mod_list}
+%{i}
+%{endfor}
+)
+
+unset(${CMAKE_FIND_PACKAGE_NAME}_FOUND)
+unset(_Botan_missing_required_modules)
+
+foreach(_comp IN LISTS ${CMAKE_FIND_PACKAGE_NAME}_FIND_COMPONENTS)
+  if (NOT _comp IN_LIST _Botan_supported_components)
+    set(${CMAKE_FIND_PACKAGE_NAME}_${_comp}_FOUND False)
+    if(${CMAKE_FIND_PACKAGE_NAME}_FIND_REQUIRED_${_comp})
+      list(APPEND _Botan_missing_required_modules ${_comp})
+    endif()
+  else()
+    set(${CMAKE_FIND_PACKAGE_NAME}_${_comp}_FOUND True)
+  endif()
+endforeach()
+
+if(_Botan_missing_required_modules)
+  set(${CMAKE_FIND_PACKAGE_NAME}_FOUND False)
+  list(JOIN _Botan_missing_required_modules ", " _missing_modules)
+  set(${CMAKE_FIND_PACKAGE_NAME}_NOT_FOUND_MESSAGE "Unsupported module(s): ${_missing_modules}")
+endif()
+
+if(DEFINED ${CMAKE_FIND_PACKAGE_NAME}_FOUND AND NOT ${${CMAKE_FIND_PACKAGE_NAME}_FOUND})
+  return()
+endif()
+
+%{if build_static_lib}
+if(NOT TARGET Botan::Botan-static)
+  add_library(Botan::Botan-static STATIC IMPORTED)
+  set_target_properties(Botan::Botan-static
+    PROPERTIES
+      IMPORTED_LOCATION                 "${CMAKE_CURRENT_LIST_DIR}/../../%{static_lib_name}"
+      INTERFACE_INCLUDE_DIRECTORIES     "${CMAKE_CURRENT_LIST_DIR}/../../../include/botan-%{version_major}"
```

Comment:
The prefix can be computed via:

```
set(_Botan_PREFIX "${CMAKE_CURRENT_LIST_DIR}")
%{for dir_component_in_destination}
get_filename_component(_Botan_PREFIX "${_Botan_PREFIX}" DIRECTORY)
%{endfor}
```

All of these `/../` traversals can then just use `${_Botan_PREFIX}` instead.